### ISSUE TITLE
Add new category in the "select category" dropdown

### ISSUE THREAD

**Issue by gwes** [johnnyq/itflow#146]
When you are in the middle of creating something like a quote, it's sometimes a bit of a pain to close out of the window you are in to go and create a new "thing" that you need before you can create the thing you were doing.

If you realise you need a new category when creating a new quote it would be good to have a quick add / add new option on the dropdown to allow you to add it without requiring clicking away.

Only a small thing and maybe once you are up and running it would not be a problem very often.

**Comment by johnnyq** [comment]
I agree completely, ill get to this Thanks @gwes

**Comment by johnnyq** [comment]
This has been added, I added it to quite a few drop downs.

Although one downside is when you add the category it closes out of the model windows your in so you must reopen the modal.

This may be fixed using an AJAX call
Some Order just brain storming

- + Clicked
- type in category
- check clicked
- submit post data via ajax to post.php
- return post reponse id and name
- inject it into DOM
- close + modal

but anyway it will still make things alot easier give it a shot let me know what you think, going to close this for now.

d2b37a89403dc3bda2d9f8d04458f599857f0739
